interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+            species_name = species.input_data["name"]
+            if "note" in species.input_data:
+                note[species_name] = species.input_data["note"]
+            else:
+                note[species_name] = ""
+
+        # sort names
+        species_names = [species.name for species in solution_species][::-1]
+        if sort_species:  # sort solution_species by molecular weight
+            MW = solution.molecular_weights
+            species_names = [c for _, c in sorted(zip(MW, species_names))]
+
+            if sort_species:
+                species_names = species_names[::-1]
+
+        # get max name length
+        max_species_len = max([len(s.name) for s in solution_species])
+
+        if any(note.values()):  # check that any notes exist
+            max_species_len = max([5, max_species_len])
+            species_txt = []
+            for species_name in species_names:
+                text = f"{species_name:<{max_species_len}}   ! {note[species_name]}"
+                species_txt.append(text)
+
+            species_txt = "\n".join(species_txt)
+
+        else:
+            species_names = [f"{name:<{max_species_len}}" for name in species_names]
+            species_names = fill(
+                "  ".join(species_names),
+                width=72,  # max length is 16, this gives 4 species per line
+                break_long_words=False,
+                break_on_hyphens=False,
+            )
+
+            species_txt = f"{species_names}"
+
+        text = ["SPECIES", f"{species_txt}", "END\n\n"]
+
+        return "\n".join(text)
+
+    text = []
+    text.append(build_elements(solution.element_names, solution.atomic_weights))
+    text.append(build_species(solution.species()))
+
+    return "\n".join(text)
+
+
+def build_thermodynamics_text(solution_species, separate_file=False):
+    """
+    Creates the thermodynamic definition text of all species
+
+    :param solution_species:
+        The species from the `solution` object being converted
+    :param separate_file:
+        A boolean flag to indicate if the file will be written separately or
+        in the mechanism file
+    """
+
+    def build_thermo_notes(solution_species):
+        """
+        Tries to infer if a note is within the NASA7 definition or is a note to be
+        placed as a comment.
+        """
+        comment = {}
+        note = {}
+        for species in solution_species:
+            if "note" in species.input_data["thermo"]:
+                note_raw = species.input_data["thermo"]["note"]
+            else:
+                note_raw = ""
+
+            # attempt to split note and comment
+            if len(note_raw.split("\n", 1)) == 1:
+                comment[species.name] = ""
+                comment_str = ""
+                note[species.name] = note_raw
+            else:
+                comment[species.name] = "!\n"
+                note[species.name], comment_str = note_raw.split("\n", 1)
```

Comment:
It'd be helpful to add a comment here with an example of what you're considering a comment vs. a note.

Reply:
Done. For clarification to you: a note is within the NASA7 text in the extra space after the name. A comment can be much longer either preceding the definition or commented next to it.